test/common/router/router_test.cc

```diff
@@ -466,8 +469,9 @@ TEST_F(RouterTest, AddMultipleCookies) {
 TEST_F(RouterTest, MetadataNoOp) { EXPECT_EQ(nullptr, router_.metadataMatchCriteria()); }
 
 TEST_F(RouterTest, MetadataMatchCriteria) {
-  MockMetadataMatchCriteria matches;
+  envoy::api::v2::core::Metadata metadata;
 
+  EXPECT_CALL(callbacks_.request_info_, dynamicMetadata()).WillRepeatedly(ReturnRef(metadata));
```

Comment:
I think you should modify the mock in test/mocks/request_info to return this by default.